Signal/src/util/MainAppContext.m

```diff
@@ -44,6 +49,23 @@ - (void)endBackgroundTask:(UIBackgroundTaskIdentifier)backgroundTaskIdentifier
     [UIApplication.sharedApplication endBackgroundTask:backgroundTaskIdentifier];
 }
 
+- (void)ensureSleepBlocking:(BOOL)shouldBeBlocking blockingObjects:(NSArray<id> *)blockingObjects;
```

Comment:
Wouldn't it be simpler to keep `DeviceSleepManager` in the main app only, forward `addBlock` and `removeBlock` using the `AppContext`, have the SAE context methods noop, done?  

I'm hoping that the `AppContext`s can remain thin "shims" that mostly do forwarding.  I recently modified the SAE context do some setup and I have mixed feelings about it.

WDYT?